pkg/nrtupdater/node_getter.go

```diff
@@ -34,27 +42,34 @@ func (ng *DisabledNodeGetter) Get(ctx context.Context, nodeName string, opts met
 	return nil, fmt.Errorf("%w", NotConfigured)
 }
 
-type OneshotNodeGetter struct {
-	K8SCli kubernetes.Interface
-	node   *corev1.Node
+type CachedNodeGetter struct {
+	nodes map[string]*corev1.Node
 }
 
-func (ng *OneshotNodeGetter) Get(ctx context.Context, nodeName string, opts metav1.GetOptions) (*corev1.Node, error) {
-	if ng.node != nil {
-		return ng.node, nil
+func NewCachedNodeGetter(k8sInterface kubernetes.Interface, ctx context.Context) *CachedNodeGetter {
+	nodelist, err := k8sInterface.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil
```

Comment:
Not completely ok with returning nil here.

I was thinking about checking in the `Get` if the map has been populated and retry again if not. 
Problem is that option would mean to check a boolean everytime, even if the map already has data, and make the first call take more time. 

In the end I kept `Get` as simple as possible and let client to handle the nil 
I have defaulted to `disable` in the resourcetopologyexporter.go